### PR TITLE
fix(getContent): fix handle un-indexed (usually private) content

### PR DESCRIPTION
**Pull request by tomwayson**
AFFECTS PACKAGES:
@esri/hub-content

ISSUES CLOSED: #341

### CODE REVIEW

File: packages/content/src/content.ts [hunk 1]
```diff
@@ -19,7 +19,12 @@ export function getContent(
   if (requestOptions && requestOptions.isPortal) {
     return getContentFromPortal(id, requestOptions);
   } else {
-    return getContentFromHub(id, requestOptions);
+    return getContentFromHub(id, requestOptions).catch(() => {
+      // TODO: inspect error?
+      // dataset is not in index (i.e. might be a private item)
+      // try fetching from portal instead
+      return getContentFromPortal(id, requestOptions);
```
Comment (drewctate) [inline packages/content/src/content.ts:26]: I don't love the fact that we use `portal` to refer both to enterprise sites and to the portal API here (and everywhere else in the app). Maybe we could consider calling this function `getContentFromPortalApi` or something. Anyway, your call.

**Comment by drewctate** [comment]
Do we need a changelog entry?

**Comment by tomwayson** [comment]
Thanks @drewctate, changelog will be _generated_ ;)

File: packages/content/src/content.ts (hunk 1 above)
Comment (tomwayson) [reply packages/content/src/content.ts:26]: Fair point. I would suggest we also rename `getContentFromHubApi` as well then.

Comment (mikeringrose) [inline packages/content/src/content.ts:26]: is this supported for all item types? if somebody is selecting a feature layer dataset, a failed request would return a feature service item. is that acceptable?

Comment (tomwayson) [reply packages/content/src/content.ts:26]: good point, but I think that is out of scope of this PR, b/c `getContent()` doesn't yet support feature layer/service datasets.

We should address this as part of #322

**Pull request merged**